src/applications/oscilloscope.py

```diff
@@ -1,117 +1,247 @@
+        else:
+            self.fig.set_size_inches(fig_width, fig_height * 1.1)
+            displayed_values = 'NaN'
+            displayed_labels = 'Cur. Val.'
+        self.current_value_text: plt.Text = self.ax.text(
+            1, 1.05, displayed_values, fontsize=self.text_font_size, transform=self.ax.transAxes, ha='right')
+        self.ax.text(
+            0, 1.05, displayed_labels, fontsize=self.text_font_size, transform=self.ax.transAxes, ha='left')
 
     def init(self):
+        """ Initializes the figure data. """
         self.line.set_ydata(self.ydata)
+        if self.show_rolling_mean:
+            self.rolling_mean_line.set_ydata(self.rolling_mean)
+        if self.show_reading_text:
+            self.current_value_text.set_text('NaN\nNaN\nNaN (NaN)' if self.show_rolling_mean else 'NaN')
         return self.line,
 
-
-    def update(self, y):
-
+    def update(self, y: Any):
+        """Updates the figure data according given a new data point."""
         self.ydata.popleft()
         self.ydata.append(y)
 
-        #this doesn't work with blit = True.
-        #there's a workaround if we need blit = true
-        #https://stackoverflow.com/questions/53423868/matplotlib-animation-how-to-dynamically-extend-x-limits
-        #need to sporadically call
-        #fig.canvas.resize_event()
+        # this doesn't work with blit = True.
+        # there's a workaround if we need blit = true
+        # https://stackoverflow.com/questions/53423868/matplotlib-animation-how-to-dynamically-extend-x-limits
+        # need to sporadically call
+        # fig.canvas.resize_event()
 
-        delta = 0.1*np.max(self.ydata)
+        delta = 0.1 * np.max(self.ydata)
         new_min = np.max([0, np.min(self.ydata) - delta])
         new_max = np.max(self.ydata) + delta
         current_min, current_max = self.ax.get_ylim()
-        if (np.abs((new_min - current_min)/(current_min)) > 0.12) or (np.abs((new_max - current_max)/(current_max)) > 0.12):
+
+        if (np.abs((new_min - current_min) / current_min) > 0.12) \
+                or (np.abs((new_max - current_max) / current_max) > 0.12):
             self.ax.set_ylim(np.max([0.01, np.min(self.ydata) - delta]), np.max(self.ydata) + delta)
+
         self.line.set_ydata(self.ydata)
-        return self.line,
 
+        if self.show_rolling_mean:
+            new_rm_value = self.update_rolling_mean()
+            self.rolling_mean_line.set_ydata(self.rolling_mean)
 
+        if self.show_reading_text:
+            text = self._get_text_value(y)
+            if self.show_rolling_mean:
+                text = f'{text}\n {self._get_text_value(new_rm_value)}'
+                measured_stdev = self.get_new_rolling_stdev_val()
+                expected_stdev = np.sqrt(new_rm_value)
+                text = f'{text}\n {self._get_text_value(measured_stdev)} ({self._get_text_value(expected_stdev)})'
+            self.current_value_text.set_text(text)
```

Comment:
I am not sure how slow it would be, I haven't tested it. At the time of writing, I thought it would be best to not call the filter, since it is a convolution function (if I remember correctly) and I believe they can get quite expensive.